### PR TITLE
Feature/refactor portal profile

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Refactored portal profile filters.
Added new portal profile filter network.

### CODE REVIEW

**Comment by extrafu** [comment]
Great work!

File: lib/pf/factory/profile/filter.pm [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+sub getData {
+    my ($class, $profile, $filter) = @_;
+    my ($type, $value);
+    if ($filter =~ m/^([^:]+):(.+)$/) {
+        $type  = $1;
+        $value = $2;
+    } else {
+        $type  = 'ssid';
```
Comment (cgx) [inline lib/pf/factory/profile/filter.pm]: Is this some default fallback? Please comment your code.

File: lib/pf/factory/profile/filter.pm [hunk 2]
```diff
@@ -0,0 +1,90 @@
+package pf::factory::profile::filter;
+
+=head1 NAME
+
+pf::factory::profile::filter add documentation
```
Comment (cgx) [inline lib/pf/factory/profile/filter.pm:5]: Please complete the documentation now!

File: lib/pf/profile/filter.pm [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package pf::profile::filter;
+=head1 NAME
+
+pf::profile::filter add documentation
```
Comment (cgx) [inline lib/pf/profile/filter.pm]: Please complete the documentation now!

File: lib/pf/profile/filter/network.pm [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package pf::profile::filter::network;
+=head1 NAME
+
+pf::profile::filter::network add documentation
```
Comment (cgx) [inline lib/pf/profile/filter/network.pm]: Yes .. do it :)

**Comment by cgx** [comment]
It would be nice to add a section in the developer's guide on how to write a new portal profile filter and list all the attributes on which a filter can be defined.

File: lib/pf/factory/profile/filter.pm [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+sub getData {
+    my ($class, $profile, $filter) = @_;
+    my ($type, $value);
+    #Split parse the filter by type and value
+    if ($filter =~ $FILTER_REGEX ) {
+        $type  = $1;
+        $value = $3;
+    } else {
+        #If there is no type defined to support older filters
```
Comment (cgx) [inline lib/pf/factory/profile/filter.pm]: Specify the PacketFence version number to which _older filters_ refer.

**Pull request merged**